### ISSUE TITLE
Explore page

### ISSUE THREAD

**Issue by nicolastakashi** [perses/perses#1453]
### Proposal

Build a Metrics Explore Page within Perses, similar to the functionality offered by Grafana and PromLens. This feature will significantly improve the usability and analytical capabilities of Perses by providing users with a dedicated interface to interactively explore and visualize metric data.

**Comment by celian-garcia** [comment]
https://github.com/perses/perses/discussions/1859

**Comment by AntoineThebaud** [comment]
Just spotted an annoying display issue when you have a graph that fits in the viewport but the pop-up on series hover doesn't: it makes the vertical scrollbar appear/disappear depending on where you point on the screen.

https://github.com/perses/perses/assets/7058693/cc4dcada-c7d2-4001-8065-4172959dd561

**Comment by celian-garcia** [comment]
We had a very interesting discussion with @AntoineThebaud on the prometheus result datamodel, and more than using a generic data table, we realized that the table displaying prometheus metrics is far more clever than that . E.g display of native histograms https://youtu.be/NDb9yZl1u1Y?t=939

There is a world where our "metrics explorer as a table" display would be as rich as the Prometheus Datatable 
https://github.com/prometheus/prometheus/blob/4b7a44c7a0568b4d9b7050b05d483fd54845d201/web/ui/react-app/src/pages/graph/DataTable.tsx#L1

If I can import it directly that would be incredible

**Comment by nicolastakashi** [comment]
> We had a very interesting discussion with @AntoineThebaud on the prometheus result datamodel, and more than using a generic data table, we realized that the table displaying prometheus metrics is far more clever than that . E.g display of native histograms https://youtu.be/NDb9yZl1u1Y?t=939
> 
> There is a world where our "metrics explorer as a table" display would be as rich as the Prometheus Datatable https://github.com/prometheus/prometheus/blob/4b7a44c7a0568b4d9b7050b05d483fd54845d201/web/ui/react-app/src/pages/graph/DataTable.tsx#L1
> 
> If I can import it directly that would be incredible

I'm just a bit worries if it couldn't be a cyclical dependency in the future, in case Prometheus decide to use Perses Components.

wdyt @Nexucis ?

**Comment by AntoineThebaud** [comment]
That's something I mentionned in our discussion actually, that we are aiming to hopefully one day be able to propose embedding Perses components in prometheus/thanos UI. So at some point instead of importing this code we'd make a copy of it in our repo, then ultimately have Prometheus importing Perses components.

**Comment by Nexucis** [comment]
> I'm just a bit worries if it couldn't be a cyclical dependency in the future, in case Prometheus decide to use Perses Components.

> wdyt @Nexucis ?

If one day it's happening, Likely Prometheus will remake the Graph page by just include the graph (so by importing the timeseries chart/plugin and the prometheus plugin).

I don't think we will have a cyclical dependency or if there is one, it will be easy to avoid. (But that's just my opinion)

**Comment by nicolastakashi** [comment]
> > I'm just a bit worries if it couldn't be a cyclical dependency in the future, in case Prometheus decide to use Perses Components.
> 
> > wdyt @Nexucis ?
> 
> If one day it's happening, Likely Prometheus will remake the Graph page by just include the graph (so by importing the timeseries chart/plugin and the prometheus plugin).
> 
> I don't think we will have a cyclical dependency or if there is one, it will be easy to avoid. (But that's just my opinion)

That's fair, I think we're good to import this right now, just raising awareness.
Besides that, looks a good solution.

**Comment by celian-garcia** [comment]
Yeah nevermind I checked a bit the prometheus code and we are too far. In terms of data model mainly. We have a data model that is probably designed to be agnostic from Prometheus. I'm wondering if it exists an agnostic data model like that in the opentelemetry specification or if it's quite free. I will check and see if we can do something closer. For example we don't support native histograms and Prometheus does. That would be cool to have it as well in the explorer.

TLDR I'm copy pasting and adapting.  
Spoiler alert: https://github.com/perses/perses/pull/1982

Btw copy paste is also what thanos did so...

**Comment by celian-garcia** [comment]
A little list of the few things that needs to be done additionally in my opinion before having a v1 ok 
- [ ] Performance consideration : 
   - [ ] limit result to 1000 in table with a warning message like prom
   - [ ] if we do a pretty formatting. Don't do the formatting if too many result as well
- [ ] Maybe put label names in bold (pretty formatting) like Prometheus is doing
- [ ] Modify Panel to give the ability to let the height grow with data (allowing to give min height / max height) That's definitely something that would be interesting for an embedded panel developer for example

**Comment by ibakshay** [comment]
Link to the Native Histogram feature request - https://github.com/perses/perses/issues/2601

**Comment by Nexucis** [comment]
I believe we can close this issue as the explorer has involved beyond the expectations and it is still growing.
It includes the traces explorer and profile explorer is coming soon.

This issue does not make sense anymore specially since explorer is a plugin now.
